src/main/java/ch/njol/skript/util/BlockUtils.java

```diff
@@ -99,4 +101,23 @@ public static Location getLocation(final @Nullable Block b) {
 		return l;
 	}
 	
+	@Nullable
+	public static BlockData createBlockData(String dataString) {
+		// Skript uses a comma to separate lists, so we use a semi colon as a delimiter
+		// Here we are just replacing it back to a comma to create a new block data
+		String data = dataString.replace(";", ",");
+		// Remove white space within square brackets ([ lit = false] -> [lit=false])
+		data = data.replaceAll(" (?=[^\\[]*])", "");
+		// If there are spaces before the square bracket we remove that
+		data = data.replace(" [", "[");
```

Comment:
I might be wrong, but wouldn't there still be white space if there were multiple spaces before the bracket?

Reply:
Fixed via all the dudes on discord:)